model/websocket_client.go

```diff
@@ -137,40 +143,40 @@ func (wsc *WebSocketClient) Listen() {
 	}()
 }
 
-func (wsc *WebSocketClient) SendMessage(action string, data map[string]interface{}) {
+func (wsc *WebSocketClient) SendMessage(action string, data map[string]interface{}) error {
```

Comment:
I hate to say this (really), but we can't break semver for `model`. The only two solutions I see is to add a new methods `SendMessage2` (Open for better suggestions :wink:) and wrap `SendMessage` or add a whole new struct `WebSocketClient2`.

Reply:
I think I'm going about this problem wrong, as this does nothing to prevent the flaky test from failing. The real issue is that the test environment unintentionally allows the sending of websocket messages to fail. Do you know how we can make the test more abstract to avoid any low level issues @hanzei?

Reply:
No. But I think the tests _should_ be low level to catch issues that arise on this level. What do you think about the wrapper method approach to debug for further debugging?

Reply:
That makes sense with the low level assertions. And the contracts with this struct need to stay as-is.

Going off of some of your ideas for solutions:

We don't have an opportunity to wrap `SendMessage` because it is not called directly in the test. It is a side effect (required side effect) for instantiating the client, and registering it with the system. Which is why the test is failing in the first place. Here's the call chain:

```
TestHelper.CreateWebsocketClient() > model.NewWebSocketClient4() > 
model.NewWebSocketClientWithDialer() > client.SendMessage()
```

We would basically have to duplicate everything, just to be able to bubble up errors. As you said, we would have to make a `WebSocketClient2` which would then need to be kept in sync with the implementation of `WebSocketClient`. I'm not sure if this change is worth the problem it is trying to solve.

Reply:
You should have `WebSocketClient` just as a wrapper `WebSocketClient2` and `WebSocketClient2` would return errors and `WebSocketClient` would just return it.

What do you think about bringing this up in Triage? I would like to hear Jesses comments about this.

The "solutions" I see are:
1. Ignore them problem and live with the flaky test
2. Break semver
3. Wrap `WebSocketClient` in something new

Reply:
4. Compare against `th.App.TotalWebsocketConnections()`

Presumably, the intent of this test isn't to verify the ability to establish a websocket connection, but that the analytics reflect reality. Perhaps if we change the test to (attempt to) create multiple connections and then compare directly against `th.App.TotalWebsocketConnections()`, we can satisfy this intent and eliminate the flakiness?

Reply:
I'm fine with this approach. But I still think we should return errors in the `WebSocketClient` methods. Maybe we create a ticket for MM 6.0?

Reply:
@lieut-data Great solution, updated with your suggestion.